### ISSUE TITLE
godot dont start on arch linux.

### ISSUE THREAD

**Issue by LinuxDonald** [godotengine/godot#6053]
**Operating system or device - Godot version:**
Arch Linux
Godot Version 2.0.4.1

**Issue description** (what happened, and what was expected):
Dont start

[linuxdonald@linuxdonald-pc ~]$ godot
==21096==AddressSanitizer CHECK failed: /build/llvm/src/llvm-3.8.1.src/projects/compiler-rt/lib/asan/asan_rtl.cc:407 "((!asan_init_is_running && "ASan init calls itself!")) != (0)" (0x0, 0x0)
    <empty stack>

[linuxdonald@linuxdonald-pc ~]$

**Comment by akien-mga** [comment]
More information would be welcome, as the official Linux binaries are not built with clang but with gcc, so I guess you are using either self-compiled binaries, or those from AUR.

**Comment by LinuxDonald** [comment]
I have used the script from AUR(self-compiled). How can I compile it with gcc? I could report it than to on the aur package.

**Comment by SajeOne** [comment]
I got the same thing when compiling with the godot-pulse AUR package. It gets the source from the 2.1 stable tarball and uses the following parameters when compiling using scons/gcc.

```
scons platform=x11 \
        tools=yes \
        target=release_debug \
        use_llvm=yes \
        builtin_zlib=yes \
        openssl=yes \
        colored=yes \
        pulseaudio=yes bits=${_arch}
```
${_arch} = the users system architecture

I then compiled using the godot-git package, which directly clones the master repo and builds from there and it works perfectly fine using the following scons parameters:

```
scons platform=x11 target=release_debug
```

I have yet to find out if one of the parameters are to blame or differences between the latest commit and 2.1 stable.

**Comment by akien-mga** [comment]
Is this issue still reproducible?

**Comment by LinuxDonald** [comment]
godot 2.1.3 dont compile on arch linux with clang 4.0

**Comment by LinuxDonald** [comment]
alot of this errors for example: thirdparty/opus/silk/macros.h:69:5: warning: macro expansion producing 'defined' has undefined behavior [-Wexpansion-to-defined]
thirdparty/opus/silk/macros.h:47:70: note: expanded from macro 'OPUS_FAST_INT64'
#define OPUS_FAST_INT64 (defined(__x86_64__) || defined(__LP64__) || defined(_WIN64))

**Comment by akien-mga** [comment]
@LinuxDonald Those are not errors but warnings, it shouldn't prevent building.

**Comment by LinuxDonald** [comment]
maybe that could help you? https://pastebin.com/j4r31XBS

**Comment by akien-mga** [comment]
See #8624.

**Comment by Calinou** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about it here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
I haven't seen any similar reports since then so I'll assume it's no longer reproducible.

If you can still reproduce the issue with the current master branch or stable releases (2.1.5 / 3.0.6), please comment here or open a new issue.
